### PR TITLE
docs: add getting started w/development guide 

### PR DESCRIPTION
**Pull request by adams0619**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Adds documentation to help with getting `cri-tools` up and running locally for development

#### Which issue(s) this PR fixes:

None 

#### Does this PR introduce a user-facing change?

```release-note
- add getting started w/development guide
```

### CODE REVIEW

**Review by saschagrunert — COMMENTED** [review]
Hey, thank you for the contribution :pray: 

May I ask you to sign the CNCF CLA first?

File: docs/developer-guide.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  ```bash
+    $ containerd config default # Remaining output truncated -- copy the output from this command
+    WARN[2020-09-18T21:37:38.987970869Z] This customized containerd is only for CI test, DO NOT use it for distribution.
+    version = 2
+    root = "/var/lib/containerd"
+    state = "/run/containerd"
+    ...
+  ```
+
+  _Background for why this is necessary: `docker` makes use of `containerd`, but when installed it has `cri` disabled, the updated config enables `cri` and allows `docker` to make use of the development `containerd/cri` that you have been installing up to this point_
```
Comment (hickeyma) [inline docs/developer-guide.md:105]: I think this is very valuable information on using containerd/cri. I feel however that it should reside in [containerd/cri dev guide](https://github.com/containerd/cri#getting-started-for-developers). Do you mind opening a PR there and updating its guide where necessary.

I feel in here it should just contain:

Follow the [containerd/cri dev guide](https://github.com/containerd/cri#getting-started-for-developers) to make and install containerd/cri runtime.

File: docs/developer-guide.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+### Running & using `cri-tools`
+
+---
+
```
Comment (hickeyma) [inline docs/developer-guide.md:110]: Remove

File: docs/developer-guide.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+After completing the above steps you should have `go` & `contianerd` binaries installed, `PATH` properly set, and the necessary `go` modules folder setup in your `$HOME` directory.
+
+Running `cri-tools` is very similar to the steps followed above
+
+- Clone / Fork repo into `$HOME/go/src/github.com/kubernetes-sigs/cri-tools`
```
Comment (hickeyma) [inline docs/developer-guide.md:115]: You would assume that someone has the repo by now, so not needed

File: docs/developer-guide.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+  _Background for why this is necessary: `docker` makes use of `containerd`, but when installed it has `cri` disabled, the updated config enables `cri` and allows `docker` to make use of the development `containerd/cri` that you have been installing up to this point_
+
+### Running & using `cri-tools`
+
+---
+
+After completing the above steps you should have `go` & `contianerd` binaries installed, `PATH` properly set, and the necessary `go` modules folder setup in your `$HOME` directory.
+
+Running `cri-tools` is very similar to the steps followed above
```
Comment (hickeyma) [inline docs/developer-guide.md:113]: I think this could be condensed down into:

Build and install `cri-tools` as follows:

**Review by hickeyma — CHANGES_REQUESTED** [review]
@adams0619 Thank you for adding this helpful documentation. I have some feedback provided inline and it is mostly to just simplify the doc by referencing other docs.

**Review by dougsland — COMMENTED** [review]
I would suggest squash the two patches.

**Comment by adams0619** [comment]
@saschagrunert Not sure why the following check is failing for my PR `cri-test`, don't think my changes should have any effect there... any help would be appreciated.

Error:
```
Summarizing 6 Failures:

[Fail] [k8s.io] Streaming runtime should support streaming interfaces [It] runtime should support portforward in host network 
/home/runner/work/cri-tools/cri-tools/src/github.com/kubernetes-sigs/cri-tools/pkg/validate/networking.go:253

[Fail] [k8s.io] Streaming runtime should support streaming interfaces [It] runtime should support portforward in host network 
/home/runner/work/cri-tools/cri-tools/src/github.com/kubernetes-sigs/cri-tools/pkg/validate/networking.go:253

[Fail] [k8s.io] Streaming runtime should support streaming interfaces [It] runtime should support portforward [Conformance] 
/home/runner/work/cri-tools/cri-tools/src/github.com/kubernetes-sigs/cri-tools/pkg/validate/networking.go:253

[Fail] [k8s.io] Streaming runtime should support streaming interfaces [It] runtime should support portforward in host network 
/home/runner/work/cri-tools/cri-tools/src/github.com/kubernetes-sigs/cri-tools/pkg/validate/networking.go:253

[Fail] [k8s.io] Streaming runtime should support streaming interfaces [It] runtime should support portforward [Conformance] 
/home/runner/work/cri-tools/cri-tools/src/github.com/kubernetes-sigs/cri-tools/pkg/validate/networking.go:253

[Fail] [k8s.io] Streaming runtime should support streaming interfaces [It] runtime should support portforward [Conformance] 
/home/runner/work/cri-tools/cri-tools/src/github.com/kubernetes-sigs/cri-tools/pkg/validate/networking.go:253

Ran 75 of 83 Specs in 335.644 seconds
FAIL! -- 73 Passed | 2 Failed | 0 Flaked | 0 Pending | 8 Skipped


Ginkgo ran 1 suite in 5m35.67118475s
Test Suite Failed
--- FAIL: TestCRISuite (335.68s)
Error:     cri_test.go:126: critest path: /usr/local/bin/critest
Error:     cri_test.go:161: Failed to run tests in parallel: exit status 1
FAIL
Error: Process completed with exit code 1.
```

File: docs/developer-guide.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+[Setup-Guide-for-Kubernetes-Developers](https://developer.ibm.com/articles/setup-guide-for-kubernetes-developers/)
+
+**Skip the `kubernetes` steps from the guide above if you don't plan on doing any development in the `kubernetes` or `crictl` projects.**
+
+---
+`cri-tools` need the following programs installed in a system in order to work correctly:
+
+- [Go](https://golang.org/doc/install)
+- [Docker](https://docs.docker.com/get-docker/)
```
Comment (saschagrunert) [inline docs/developer-guide.md:15]: For which part do we need Docker? I doubt that we require it.

File: docs/developer-guide.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+## Pre-requisites
+
+Start w/the following setup guide:
+
+[Setup-Guide-for-Kubernetes-Developers](https://developer.ibm.com/articles/setup-guide-for-kubernetes-developers/)
+
+**Skip the `kubernetes` steps from the guide above if you don't plan on doing any development in the `kubernetes` or `crictl` projects.**
+
+---
+`cri-tools` need the following programs installed in a system in order to work correctly:
```
Comment (saschagrunert) [inline docs/developer-guide.md:12]: I would rephrase it to mention that we need those for development only.

File: docs/developer-guide.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    PATH=$PATH:/usr/local/go/bin:$HOME/go/bin # $PATH here refers to truncated version of additional `env` paths that are unrelated this guide / setup
+  ```
+
+## Build & install `containerd/cri`
+
+If you followed the guide so far you may likely already have `containerd` installed, but we will need to install the latest development version.
+
+Running the development versions of `containerd` is not required if you do not plan on developing for `crictl`, `containerd` or `kubernetes`, you can instead install the latest packaged version for your given OS... i.e. for `Ubuntu` run `apt install contianerd`
+
+This guide will use `containerd/cri` as the container runtime. You may use `cri-o` instead but you may need some additional steps to ensure it works. You can also follow the `cri-o` [install guide](https://github.com/cri-o/cri-o/blob/master/install.md#install-packaged-versions-of-cri-o) if you prefer to use the packaged version.
```
Comment (saschagrunert) [inline docs/developer-guide.md:38]:
```suggestion
This guide will use `containerd/cri` as the container runtime. You may use CRI-O instead but you may need some additional steps to ensure it works. You can also follow the CRI-O [install guide](https://github.com/cri-o/cri-o/blob/master/install.md#install-packaged-versions-of-cri-o) if you prefer to use the packaged version.
```

**Comment by saschagrunert** [comment]
Yes the test failures are not coming from this patch, I'll check what's going on.

**Comment by saschagrunert** [comment]
Looks like a flake, not it's working :)

File: docs/developer-guide.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    $ ls $HOME
+    go
+
+    $ echo $PATH | grep go
+    PATH=$PATH:/usr/local/go/bin:$HOME/go/bin # $PATH here refers to truncated version of additional `env` paths that are unrelated this guide / setup
+  ```
+
+## Build & install `containerd/cri`
+
+If you followed the guide so far you may likely already have `containerd` installed, but we will need to install the latest development version.
```
Comment (hickeyma) [inline docs/developer-guide.md:34]: If you followed the guide so far you may likely already have `containerd` installed, but we will need to install the latest development version. --> You will require the latest development version of `containerd`.

File: docs/developer-guide.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+Running the development versions of `containerd` is not required if you do not plan on developing for `crictl`, `containerd` or `kubernetes`, you can instead install the latest packaged version for your given OS... i.e. for `Ubuntu` run `apt install contianerd`
```
[the remark this answers is not in the document]
Comment (hickeyma) [reply docs/developer-guide.md:36]: Not sure this sentence is needed as it seems to conflict with the previous sentence?

File: docs/developer-guide.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+This guide will use `containerd/cri` as the container runtime. You may use `cri-o` instead but you may need some additional steps to ensure it works. You can also follow the `cri-o` [install guide](https://github.com/cri-o/cri-o/blob/master/install.md#install-packaged-versions-of-cri-o) if you prefer to use the packaged version.
+
+- Follow the [`containerd/cri` dev guide](https://github.com/containerd/cri#getting-started-for-developers) to `make` and install `containerd/cri` runtime.
```
Comment (hickeyma) [inline docs/developer-guide.md:40]: install --> install the

File: docs/developer-guide.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+## Build & install `cri-tools`
+
+You can build and install `cri-tools` following the steps below:
```
Comment (hickeyma) [inline docs/developer-guide.md:44]: You can build and install `cri-tools` following the steps below: --> You can build and install `cri-tools` as follows:

File: docs/developer-guide.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+
+- Install necessary build tools using `make` within the repo directory
```
Comment (hickeyma) [inline docs/developer-guide.md:46]: Install necessary build tools using `make` within the repo directory --> Install the dependencies:

File: docs/developer-guide.md [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+
+  ```bash
+    $ make install.tools
+  ```
+
+- Build `cri-tools` from source and install its binary using `make`
```
Comment (hickeyma) [inline docs/developer-guide.md:52]: Build `cri-tools` from source and install its binary using `make` --> Build `cri-tools` (`critest` and `crictl`) and install to common location:

**Review by hickeyma — CHANGES_REQUESTED** [review]

**Comment by hickeyma** [comment]
Thanks for updating @adams0619. Nearly there, just some more comments inline.

**Comment by adams0619** [comment]
@hickeyma @saschagrunert Updated to address comments... could I get another review on this?

File: docs/developer-guide.md [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+    $ ls $HOME
+    go
+
+    $ echo $PATH | grep go
+    PATH=$PATH:/usr/local/go/bin:$HOME/go/bin # $PATH here refers to truncated version of additional `env` paths that are unrelated this guide / setup
+  ```
+
+## Build & install `containerd/cri`
+
+The latest development version of `containerd` is required.
```
Comment (saschagrunert) [inline docs/developer-guide.md:32]:
```suggestion
The latest development version of `containerd` or CRI-O is required.
```

**Review by saschagrunert — APPROVED** [review]
Just a nit, otherwise LGTM
/approve

File: docs/developer-guide.md [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+
+Running the development versions of `containerd` is not required if you do not plan on developing for `containerd` or `kubernetes`, you can instead install the latest packaged version for your given OS... i.e. for `Ubuntu` run `apt install contianerd`
```
Comment (hickeyma) [inline docs/developer-guide.md:34]: Remove as this is confusing with the sentence before it.

File: docs/developer-guide.md [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+
+This guide will use `containerd/cri` as the container runtime. You may use CRI-O instead but you may need some additional steps to ensure it works. You can also follow the CRI-O [install guide](https://github.com/cri-o/cri-o/blob/master/install.md#install-packaged-versions-of-cri-o) if you prefer to use the packaged version.
+
+- Follow the [`containerd/cri` dev guide](https://github.com/containerd/cri#getting-started-for-developers) to `make` and install the `containerd/cri` runtime.
```
Comment (hickeyma) [inline docs/developer-guide.md:38]: Remove `-` because it should not be bulleted.

**Review by hickeyma — APPROVED** [review]
2 small nits, otherwise LGTM.

**Review by hickeyma — APPROVED** [review]
LGTM, thanks for your effort on this @adams0619

**Review by saschagrunert — APPROVED** [review]
/lgtm
